crates/red_knot_module_resolver/src/path.rs

```diff
@@ -408,7 +409,7 @@ impl SearchPath {
                 typeshed.to_path_buf(),
             ));
         }
-        let Some(typeshed_versions) = system_path_to_file(db.upcast(), stdlib.join("VERSIONS"))
+        let Ok(typeshed_versions) = system_path_to_file(db.upcast(), stdlib.join("VERSIONS"))
         else {
             return Err(SearchPathValidationError::NoVersionsFile(typeshed));
```

Comment:
I suppose we could take the opportunity here to give a custom error message if `stdlib/VERSIONS` is a directory rather than a file, now that we have that information with this refactor